qttypes/src/lib.rs

```diff
@@ -875,6 +883,336 @@ mod tests {
+            return self->isEmpty();
+        })
+    }
+
+    /// Wrapper around [`contains(const QString &)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qmap.html#contains
+    pub fn contains(&self, key: QString) -> bool {
+        cpp!(unsafe [self as "QVariantMap*", key as "QString"] -> bool as "bool" {
+            return self->contains(key);
+        })
+    }
+
+    /// Wrapper around [`clear()`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qmap.html#clear
+    pub fn clear(&mut self) {
+        cpp!(unsafe [self as "QVariantMap*"] {
+            self->clear();
+        })
+    }
+
+    /// Wrapper around [`value(const QString &, const QVariant &)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qmap.html#value
+    pub fn value(&self, key: QString, default_value: QVariant) -> QVariant {
+        cpp!(unsafe [self as "QVariantMap*", key as "QString", default_value as "QVariant"] -> QVariant as "QVariant" {
+            return self->value(key, default_value);
+        })
+    }
+
+    /// Wrapper around [`key(const QVariant &, const QString &)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qmap.html#key
+    pub fn key(&self, value: QVariant, default_key: QString) -> QString {
+        cpp!(unsafe [self as "QVariantMap*", default_key as "QString", value as "QVariant"] -> QString as "QString" {
+            return self->key(value, default_key);
+        })
+    }
+}
+
+impl Index<QString> for QVariantMap {
+    type Output = QVariant;
+
+    /// Wrapper around [`at(int)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qlist.html#at
+    #[track_caller]
+    fn index(&self, key: QString) -> &QVariant {
+        unsafe {
+            cpp!([self as "QVariantMap*", key as "QString"] -> *const QVariant as "const QVariant*" {
+                auto x = self->constFind(key);
+                if (x == self->constEnd()) {
+                    return NULL;
+                } else {
+                    return &x.value();
+                }
+            }).as_ref()
+        }.expect("key not in the QVariant")
+    }
+}
+impl IndexMut<QString> for QVariantMap {
+    /// Wrapper around [`operator[](int)`][method] operator method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qlist.html#operator-5b-5d
+    fn index_mut(&mut self, key: QString) -> &mut QVariant {
+        unsafe {
+            &mut *cpp!([self as "QVariantMap*", key as "QString"] -> *mut QVariant as "QVariant*" {
+                return &(*self)[key];
+            })
+        }
+    }
+}
+
+impl fmt::Debug for QVariantMap {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_map().entries(self.into_iter()).finish()
+    }
+}
+
+cpp_class!(unsafe struct QVariantMapIteratorInternal as "QVariantMap::iterator");
+
+/// Internal class used to iterate over a [`QVariantMap`][]
+///
+/// [`QVariantMap`]: ./struct.QVariantMap.html
```

Comment:
```suggestion
/// Internal class used to iterate over a [`QVariantMap`]
```

We can use the intra-doc links.